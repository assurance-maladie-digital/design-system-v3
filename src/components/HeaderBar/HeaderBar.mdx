import { Controls, Canvas, Meta, Source } from "@storybook/blocks";
import * as HeaderBarStories from "./HeaderBar.stories";

<Meta of={HeaderBarStories} />


Pour voir l'utilisation du menu, veuillez consulter le composant `HeaderBurgerMenu` dans le menu de gauche.

# HeaderBar

<Canvas of={HeaderBarStories.Default} />

## API

<Controls of={HeaderBarStories.Default} />

## Les slots

Le composant `HeaderBar` possède les slots suivants :

### Prepend

Le slot `prepend` permet d'ajouter du contenu en haut du header, il suivra le comportement du header au scroll.
Il est possible de récupérer le status du Menu de header en utilisant la prop `isMenuOpen` du slot `prepend`. (Fonctionne uniquement pour les menu s'enregistrant via la fonction injectée `registerHeaderMenu`, ce qui est le cas pour le composant `HeaderBurgerMenu`)

<Source
  dark
  code={`
<template>
  <HeaderBar>
	<template #menu>
	  <HeaderBurgerMenu>
		...
	  </HeaderBurgerMenu>
	</template>
	<template #prepend="{ isMenuOpen }">
	  <div v-if="isMenuOpen">Menu ouvert</div>
	  <div v-else>Menu fermé</div>
	</template>
  </HeaderBar>
</template>
`}
/>

### Append

Le slot `append` permet d'ajouter du contenu en bas du header, il suivra le comportement du header au scroll.
Il est également possible de récuperer le status du Menu de header en utilisant la prop `isMenuOpen` du slot `append`.

<Source
  dark
  code={`
<template>
  <HeaderBar>
	<template #menu>
	  <HeaderBurgerMenu>
		...
	  </HeaderBurgerMenu>
	</template>
	<template #append="{ isMenuOpen }">
	  <div v-if="isMenuOpen">Menu ouvert</div>
	  <div v-else>Menu fermé</div>
	</template>
  </HeaderBar>
</template>
`}
/>

### Menu

Le slot `menu` permet d'ajouter un menu au header. Il est recommandé d'utiliser le composant `HeaderBurgerMenu` du design system Pour tout menu contenant plus de 5 éléments.

<Source
  dark
  code={`
<template>
  <HeaderBar>
	<template #menu>
	  <HeaderBurgerMenu>
		...
	  </HeaderBurgerMenu>
	</template>
  </HeaderBar>
</template>
`}
/>

### Logo

Le slot `logo` permet d'ecraser le logo par défaut du header. Il est déconseillé de l'utiliser. Si cela n'est pas possible l'intégration d'un nouveau logo doit se faire en respectant les guidelines de la charte graphique.

<Source
  dark
  code={`
<template>
  <HeaderBar>
	<template #logo>
	  <img src="https://via.placeholder.com/150" alt="Logo" />
	</template>
  </HeaderBar>
</template>
`}
/>

### HeaderSide

Le slot `header-side` permet d'ajouter du contenu à droite du header. Il est possible d'ajouter plusieurs éléments dans ce slot.
Il sera principalement utilisé pour ajouter un boutton de connexion, d'accès à un compte ou de recherche.
Il est necessaire de faire attention à la place que prendront les éléments en mode mobile et de prévoir un mode réduit ou bien de les déporter dans le menu burger.

<Canvas
	of={HeaderBarStories.WithRightMenu}
	source={{
		language: 'html',
		format: 'dedent',
		code:`
	<script setup lang="ts">
		import { mdiMagnify, mdiAccountCircleOutline } from '@mdi/js'
	</script>

	<template>
		<HeaderBar>
			<template #header-side>
			<div class="d-flex justify-center h-100 ga-4 pr-4">
				<VBtn
					variant="text"
					:prepend-icon="mdiMagnify"
					color="primary"
				>
					Rechercher
				</VBtn>
				<VBtn
					color="primary"
					:prepend-icon="mdiAccountCircleOutline"
				>
					Login
				</VBtn>
			</div>
			</template>
		</HeaderBar>
	</template>
	`}}
/>
