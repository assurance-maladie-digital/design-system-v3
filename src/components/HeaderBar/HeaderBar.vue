<script setup lang="ts">
	import CNAMLogo from './CNAMLogo.vue'
	import { ref, computed, onMounted, onUnmounted } from 'vue'

	defineSlots<{
		'default': () => void
		'prepend': () => void
		'append': () => void
		'menu': () => void
		'logo': () => void
		'header-side': () => void
	}>()

	const props = withDefaults(defineProps<{
			sticky?: boolean
		}>(),
		{
			sticky: true,
		})

	const header = ref<HTMLElement | null>(null)
	const headerStyle = computed<{
		position: 'fixed' | 'relative'
	}>(() => {
		return {
			position: !scrollIsOnTop.value && props.sticky ? 'fixed' : 'relative',
		}
	})

	const scrollIsOnTop = ref(true)
	function handleScroll() {
		scrollIsOnTop.value = window.scrollY < headerOffset.value
	}

	const headerOffset = ref(1)
	function setHeaderOffset() {
		headerOffset.value = header.value!.getBoundingClientRect().top
	}
	onMounted(() => {
		setHeaderOffset()
		window.addEventListener('scroll', handleScroll)
		window.addEventListener('resize', setHeaderOffset)
	})

	onUnmounted(() => {
		window.removeEventListener('scroll', handleScroll)
		window.removeEventListener('resize', setHeaderOffset)
	})

</script>

<template>
	<header
		ref="header"
		class="header"
		:style="headerStyle"
	>
		<div
			v-if="$slots.prepend"
			class="header-prepend"
		>
			<slot name="prepend" />
		</div>
		<div class="inner-header d-flex">
			<!---->
			<slot
				name="menu"
			/>

			<!---->

			<div class="header-logo">
				<slot
					name="logo"
				>
					<CNAMLogo />
				</slot>
			</div>

			<div
				v-if="$slots.headerSide"
				class="header-side"
			>
				<slot name="header-side" />
			</div>
		</div>
		<div
			v-if="$slots.append"
			class="header-append"
		>
			<slot name="append" />
		</div>
	</header>
</template>

<style lang="scss" scoped>

.header {
	top: 0;
	width: 100%;
	background-color: #fff;
	border-bottom: solid 1px #ced9eb;
}

.inner-header {
	display: flex;
	align-items: center;
}

.header-side {
	display: flex;
	align-items: center;
	margin-left: auto;
}
</style>
