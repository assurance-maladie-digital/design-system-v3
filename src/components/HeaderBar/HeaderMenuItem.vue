<script setup lang="ts">

</script>

<template>
	<li class="header-menu-item">
		<slot />
	</li>
</template>

<style lang="scss" scoped>
	@use "@/assets/tokens.scss" as *;

	.header-menu-item {
		color: $primary-base;
		list-style-type: none;
		margin: 0;
		padding: 0;
		min-height: 44px; // accessibility requirement
		font-weight: 700;
		text-transform: capitalize;

		> :deep(a) {
			display: block;
			padding: 16px 50px 16px 20px;
			text-decoration: none;

			&:hover {
				text-decoration: underline;
			}

			&:focus, &:active, &:visited {
				color: currentColor;
			}
		}
	}

	.header-menu-item:hover {
		background-color: $primary-base;
		color: $neutral-white;
	}
</style>
