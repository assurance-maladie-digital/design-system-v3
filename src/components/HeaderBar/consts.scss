$header-breakpoint: 990px;
$header-height-desktop: 95px;
$header-height: 77px;
$header-max-width: 1712px;
$menu-border-color: #e0e0e0;
$menu-width: 350px;