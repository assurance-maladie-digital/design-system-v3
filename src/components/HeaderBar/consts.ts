export const headerBreakpoint = 990
export const registerHeaderMenuKey = Symbol('registerHeaderMenu')
