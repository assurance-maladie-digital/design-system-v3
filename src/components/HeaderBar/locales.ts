export const locales = {
	homeAriaLabel: 'Logo de l\'Assurance Maladie, cliquez pour revenir Ã  l\'accueil',
}
