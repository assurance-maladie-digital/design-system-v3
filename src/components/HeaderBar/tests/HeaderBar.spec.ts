import { vuetify } from '@tests/unit/setup'
import { mount } from '@vue/test-utils'
import { describe, expect, it, vi, afterEach } from 'vitest'
import { defineComponent, inject, onMounted, ref, type Ref } from 'vue'
import HeaderBar from '../HeaderBar.vue'

describe('HeaderBar', () => {
	afterEach(() => {
		vi.restoreAllMocks()
		document.body.innerHTML = ''
	})

	it('should render the component', async () => {
		const wrapper = mount(HeaderBar, {
			props: {
				sticky: false,
			},
			global: {
				plugins: [vuetify],
			},
			attachTo: document.body,
		})
		window.dispatchEvent(new CustomEvent('scroll'))

		expect(wrapper.html()).toMatchSnapshot()

		// mock scroll related values
		vi.spyOn(window, 'scrollY', 'get').mockReturnValue(2000)
		const header = wrapper.find('.header')
		vi.spyOn(header.element, 'getBoundingClientRect').mockReturnValue({
			top: -2000,
		} as DOMRect)
		window.dispatchEvent(new CustomEvent('scroll'))

		await wrapper.vm.$nextTick()

		const stickyHeader = wrapper.find('.sticky-header')
		const stickyHeaderStyle = stickyHeader.attributes('style')
		expect(stickyHeaderStyle).toContain('position: relative;')
		expect(stickyHeaderStyle).toContain('top: auto;')

		wrapper.unmount()
	})

	it('should render all the component slots', async () => {
		const wrapper = mount(HeaderBar, {
			global: {
				plugins: [vuetify],
			},
			attachTo: document.body,
			slots: {
				'default': '<div>Default slot</div>',
				'prepend': '<div>Prepend slot</div>',
				'append': '<div>Append slot</div>',
				'menu': '<div>Menu slot</div>',
				'logo': '<div>Logo slot</div>',
				'header-side': '<div>Header side slot</div>',
			},
		})

		const text = wrapper.text()

		expect(text).not.toContain('Default slot')
		expect(text).toContain('Prepend slot')
		expect(text).toContain('Append slot')
		expect(text).toContain('Menu slot')
		expect(text).toContain('Logo slot')
		expect(text).toContain('Header side slot')

		wrapper.unmount()
	})


	const TestMenu = defineComponent({
		setup() {
			const menu = ref(false)
			const registerHeaderMenu = inject<(r: Ref<boolean>) => void>('registerHeaderMenu')

			onMounted(() => {
				if (registerHeaderMenu) {
					registerHeaderMenu(menu)
				}
			})

			return {
				menu,
			}
		},
		template: '<button @click="menu = true">Menu</button>',
	})

	it('should render the menu slot', async () => {
		const wrapper = mount({
			components: {
				HeaderBar,
				TestMenu,
			},
			template: `
				<HeaderBar>
					<template #menu>
						<TestMenu />
					</template>
					<template #logo="{menuOpen}">
						<div>{{menuOpen ? 'the menu is open' : 'the menu is closed'}}</div>
					</template>
				</HeaderBar>`,
		}, {
			global: {
				plugins: [vuetify],
			},
			attachTo: document.body,
		})
		const text = wrapper.text()
		expect(text).toContain('Menu')
		expect(text).toContain('the menu is closed')

		const button = wrapper.find('button')
		await button.trigger('click')

		expect(wrapper.text()).toContain('the menu is open')

		wrapper.unmount()
	})

	it('should render in sticky mode', async () => {
		const wrapper = mount(HeaderBar, {
			props: {
				sticky: true,
			},
			global: {
				plugins: [vuetify],
			},
			attachTo: document.body,
		})
		window.dispatchEvent(new CustomEvent('scroll'))

		// mock scroll related values
		vi.spyOn(window, 'scrollY', 'get').mockReturnValue(2000)
		const header = wrapper.find('.header')
		vi.spyOn(header.element, 'getBoundingClientRect').mockReturnValue({
			top: -2000,
		} as DOMRect)
		window.dispatchEvent(new CustomEvent('scroll'))

		await wrapper.vm.$nextTick()

		const stickyHeader = wrapper.find('.sticky-header')
		const stickyHeaderStyle = stickyHeader.attributes('style')
		expect(stickyHeaderStyle).toContain('position: fixed;')
		expect(stickyHeaderStyle).toContain('top: 0px;')

		wrapper.unmount()
	})

	it('should only show the header when the user scrolls up', async () => {
		// @ts-expect-error  - Property 'happyDOM' does not exist on type 'Window & typeof globalThis'.
		window.happyDOM.setInnerWidth(600)
		const wrapper = mount(HeaderBar, {
			props: {
				hideWhenDown: true,
			},
			global: {
				plugins: [vuetify],
			},
			attachTo: document.body,
		})
		window.dispatchEvent(new CustomEvent('scroll'))

		const header = wrapper.find('.header')
		const mockHeaderRect = vi.spyOn(header.element, 'getBoundingClientRect')

		// mock scroll related values
		const mockScrollY = vi.spyOn(window, 'scrollY', 'get')
		const mockScrollTop = vi.spyOn(document.documentElement, 'scrollTop', 'get')
		mockScrollY.mockReturnValue(2000)
		mockScrollTop.mockReturnValue(2000)
		mockHeaderRect.mockReturnValue({
			top: -2000,
			height: 77,
		} as DOMRect)
		window.dispatchEvent(new CustomEvent('scroll'))
		await wrapper.vm.$nextTick()

		const stickyHeader = wrapper.find('.sticky-header')
		let stickyHeaderStyle = stickyHeader.attributes('style')

		// Do not show the header when the user scrolls down
		expect(stickyHeaderStyle).toContain('position: fixed;')
		expect(stickyHeaderStyle).toContain('top: 0px;')
		expect(stickyHeaderStyle).toContain('transform: translateY(-100%);')

		mockScrollY.mockReturnValue(800)
		mockScrollTop.mockReturnValue(800)
		mockHeaderRect.mockReturnValue({
			top: -800,
			height: 77,
		} as DOMRect)
		window.dispatchEvent(new CustomEvent('scroll'))
		await wrapper.vm.$nextTick()

		stickyHeaderStyle = stickyHeader.attributes('style')

		// Do show the header when the user scrolls up
		expect(stickyHeaderStyle).toContain('position: fixed;')
		expect(stickyHeaderStyle).toContain('top: 0px;')
		expect(stickyHeaderStyle).toContain('transform: none;')

		wrapper.unmount()
	})
})
