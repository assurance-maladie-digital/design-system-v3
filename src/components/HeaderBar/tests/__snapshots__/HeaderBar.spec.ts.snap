// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderBar > should render the component 1`] = `
"<header data-v-efdbe191="" class="header" style="min-height: auto;">
  <div data-v-efdbe191="" class="sticky-header" style="position: relative; width: 100%; top: auto; transform: none; transition: none;">
    <!--v-if-->
    <div data-v-efdbe191="" class="inner-header d-flex">
      <div data-v-efdbe191="" class="header-logo">
        <div data-v-a3ce97de="" data-v-efdbe191="" class="logo"><svg data-v-d1ac2910="" data-v-a3ce97de="" fill="#0c419a" width="141" height="42" viewBox="0 0 211 64" role="img" focusable="false" xmlns="http://www.w3.org/2000/svg" aria-label="Logo de l'Assurance Maladie, cliquez pour revenir Ã  l'accueil">
            <g data-v-d1ac2910="">
              <path d="M100.1 45V30.6h3.8v14.6c0 .8.4 1 .7 1h.5l.5 2.9a5 5 0 0 1-2 .3c-2.6 0-3.5-1.7-3.5-4.4Z" data-v-d1ac2910=""></path>
              <path d="m75.7 40.4 3.8-8.5h4V49h-3.6V38.4l-3 7.8h-2.4l-3-7.8v10.7H68V32h4l3.8 8.5Z" data-v-d1ac2910=""></path>
              <path d="M107.1 45.5c0-2.8 2.3-4.3 7.4-4.9 0-1.1-.8-1.8-2.2-1.8-1 0-2.1.4-3.4 1.2l-1.3-2.6c1.6-1 3.5-1.7 5.5-1.7 3.3 0 5 1.9 5 5.9V49h-3.5v-1.3s-1.5 1.6-3.6 1.6c-2.4 0-3.9-1.7-3.9-4Zm7.4-.3V43c-2.7.3-3.7 1.2-3.7 2.2 0 .8.5 1.2 1.5 1.2.8 0 1.5-.5 2.2-1.2Z" data-v-d1ac2910=""></path>
              <path d="M86.1 45.5c0-2.8 2.3-4.3 7.4-4.9 0-1.1-.8-1.8-2.2-1.8-1 0-2.1.4-3.4 1.2l-1.3-2.6c1.6-1 3.5-1.7 5.5-1.7 3.3 0 5 1.9 5 5.9V49h-3.5v-1.3S92 49.4 90 49.4c-2.4 0-3.9-1.7-3.9-4Zm7.4-.3V43c-2.7.3-3.7 1.2-3.7 2.2 0 .8.6 1.2 1.5 1.2.8 0 1.6-.5 2.2-1.2Z" data-v-d1ac2910=""></path>
              <path d="M129 30.6v6.3a4.1 4.1 0 0 0-3.2-1.2c-2.7 0-5.3 2.6-5.3 6.8 0 4.3 2 7 5.2 7 1.8 0 3.1-1.4 3.2-1.5v1.1h3.8V30.6H129Zm-2.3 15.7c-1.5 0-2.4-1.2-2.4-3.8 0-2.5 1.1-3.6 2.4-3.6.6 0 1.4.2 2.2.8v5.4c-.7.8-1.4 1.2-2.2 1.2Z" data-v-d1ac2910=""></path>
              <path d="M135.7 32c0-1.2 1-2 2.2-2 1.2 0 2.2.8 2.2 2s-1 2-2.2 2c-1.3 0-2.2-.8-2.2-2Zm.3 4h3.8V49H136V36Z" data-v-d1ac2910=""></path>
              <path d="M142.2 42.5c0-4.2 2.6-6.8 5.9-6.8 3.6 0 5.6 2.7 5.6 6.3l-.1 1.7h-7.7c.3 2 1.6 2.8 3.3 2.8 1 0 1.8-.4 2.8-1l1.3 2.5c-1.3.9-3 1.4-4.6 1.4-3.8 0-6.5-2.5-6.5-6.9Zm8.3-1.4c0-1.4-.9-2.5-2.4-2.5-1.2 0-2 .9-2.3 2.5h4.7Z" data-v-d1ac2910=""></path>
              <path d="M68 23.3V8.8h3.7v14.7c0 .8.4 1 .7 1h.5l.4 2.9a5 5 0 0 1-1.9.3c-2.6 0-3.5-1.7-3.5-4.4Z" data-v-d1ac2910=""></path>
              <path d="m88.1 23.3 1 4.1h4l-5.3-17.3h-4.4l-5.3 17.3h4l1-4.1h5Zm-4.2-3 1.6-6.6h.2l1.6 6.5H84Z" data-v-d1ac2910=""></path>
              <path d="m93.9 25.9 1.7-2.4c1.1.9 2.2 1.3 3.3 1.3 1.1 0 1.6-.4 1.6-1 0-1-1.2-1.4-2.6-1.9-1.6-.6-3.4-1.7-3.4-3.8 0-2.5 2-4.2 5-4.2 1.9 0 3.4.8 4.5 1.7l-1.7 2.4c-1-.7-1.8-1.2-2.8-1.2-1 0-1.4.4-1.4 1 0 1 1.2 1.2 2.5 1.7 1.6.7 3.5 1.6 3.5 4s-1.9 4.2-5.3 4.2c-1.7 0-3.6-.7-5-1.8Z" data-v-d1ac2910=""></path>
              <path d="m105.3 25.9 1.7-2.4c1.1.9 2.2 1.3 3.3 1.3 1.1 0 1.6-.4 1.6-1 0-1-1.2-1.4-2.6-1.9-1.6-.6-3.4-1.7-3.4-3.8 0-2.5 2-4.2 5-4.2 1.9 0 3.4.8 4.5 1.7l-1.7 2.4c-1-.7-1.8-1.2-2.8-1.2-1 0-1.4.4-1.4 1 0 1 1.2 1.2 2.5 1.7 1.6.7 3.5 1.6 3.5 4s-1.9 4.2-5.3 4.2c-1.7 0-3.6-.7-5-1.8Z" data-v-d1ac2910=""></path>
              <path d="M132.2 14.3h3.8V16c.8-1.4 2.5-2 3.4-2 .8 0 1.2.2 1.6.3l-.4 3.6c-.5-.2-1.1-.4-1.7-.4-1.5 0-2.8 1.3-2.9 3v6.8h-3.8V14.3Z" data-v-d1ac2910=""></path>
              <path d="M155.7 14.3h3.8v1.6c1-1 2.2-2 4-2 2.7 0 4 2 4 5.3v8.2h-3.9v-7.7c0-1.9-.5-2.5-1.6-2.5-1 0-1.6.5-2.5 1.3v8.8h-3.8v-13Z" data-v-d1ac2910=""></path>
              <path d="M180.6 20.8c0-4.3 2.7-6.9 5.9-6.9 3.7 0 5.7 2.8 5.7 6.3 0 .7 0 1.4-.2 1.8h-7.7c.4 1.9 1.7 2.8 3.4 2.8 1 0 1.8-.4 2.7-1l1.3 2.4c-1.3 1-3 1.5-4.6 1.5-3.7 0-6.5-2.5-6.5-6.9Zm8.4-1.4c0-1.4-.9-2.5-2.4-2.5-1.2 0-2 .8-2.3 2.5h4.7Z" data-v-d1ac2910=""></path>
              <path d="M141.8 23.7c0-2.7 2.3-4.3 7.4-4.8 0-1.2-.8-1.9-2.2-1.9a7 7 0 0 0-3.4 1.2l-1.3-2.5c1.6-1 3.4-1.8 5.5-1.8 3.3 0 5 2 5 6v7.5h-3.6V26s-1.4 1.6-3.5 1.6c-2.4 0-4-1.7-4-4Zm7.4-.2v-2.3c-2.7.4-3.8 1.3-3.8 2.3 0 .8.6 1.2 1.5 1.2s1.7-.5 2.3-1.2Z" data-v-d1ac2910=""></path>
              <path d="M76.7 8.8c-.8 0-1.9.6-1.9 1.7 0 1 .8 1.5 1.6 1.5 0 1.2-1 1.8-1.5 2.1l1 1.4c3.3-1 4-6.7.8-6.7Z" data-v-d1ac2910=""></path>
              <path d="M125.3 14.3V23c-.8 1-1.4 1.4-2.3 1.4-1 0-1.6-.6-1.6-2.5v-7.6h-3.8v8.3c0 3 1.4 5.1 4 5.1 2.5 0 3.8-1.6 3.8-1.6v1.3h3.7V14.3h-3.8Z" data-v-d1ac2910=""></path>
              <path d="M176.3 24.6c-1.7 0-3-1.5-3-3.7 0-2.3 1.3-3.8 3-3.8.6 0 1.1.2 1.7.7l1.8-2.5c-.9-.8-2.1-1.3-3.7-1.3a6.5 6.5 0 0 0-6.6 6.9c0 4.3 2.7 6.8 6.3 6.8 1.4 0 3-.4 4.2-1.5l-1.4-2.7c-.7.6-1.5 1-2.3 1Z" data-v-d1ac2910=""></path>
            </g>
            <path d="M34.8 9c-.1-.2-.4-.3-.6-.3H34c-.2-.2-5.3 0-6.5.7 0-.8-.4-1.8-1.7-2-1-.3-3.1.7-2.4 2.8 0 .3.3.6.7.8.2 0 .5 0 .7.6h.3l.2.2h.5c.4.4.8-.1 1-.5l.7 2c-.2-.2-1.8-.6-2.5.2s-1.2 1.8-1.7 3.5c-.2.4-.5.7-1 .9-.9.3-.4.7-.2.7a6 6 0 0 0 2 0c.4-.2.1-1 .5-1.4l.8-1c0 .5 0 .8-.3 1.5-.2.7-1 1-1.3 1.3-.4.5.6.5.8.5.7-.2 1-.2 1.3-.4.4-.2.2-.8.3-1 .3-.7 1.4-2 1.6-2.3.8.2 1.7.2 2 .2 1.9-.2 2.3-1.5 2-2.6-.1-1-.7-1.5-.9-1.8l-.9-1.4 3.5-.9c0 .3.3.3.4.4.2.1.3 0 .2-.1a.6.6 0 0 1-.2-.2h.7c.3 0 .4-.4.3-.5Z" data-v-d1ac2910=""></path>
            <path fill-rule="evenodd" clip-rule="evenodd" d="M61 26.7c-.5-2.6-2.9-4.1-4.1-4.7l-1.4-.4-.4-.2c.1-.5.8-.5 1.2-1.3 1-2.3-1.2-3.8-2.8-3.5-.8.2-1.3.4-1.5 1 0 .4-.3.4-.7.8-.2.2.1.4.2.5l-.1.3v.5s-.3.3 0 .7l.3.8c0 .1 0 .2-.3 0l-3.4-1a37.4 37.4 0 0 1-4.2-4.7c-.7 0 .5 1.3-.1.8-.2 0-.7-.4-.8-.5l-.1-.2-.3-.3H42.2c-.2 0-.1.5 0 .7 0 0 .4 1.2.6 1.4l1.5 1 2.6 3c.7.7 3.8 2.5 3.9 2.6.2 0 .5.8 0 5.5-.1 1.8-1.4 5.6-.8 7.5.4 1.3.9 2.2 1.3 5.6.1.8-.5 1.3-1.3 2l-1.4.8c-.2.2 0 .4.2.5.2 0 .5.3.8.3.3.1.9 0 1.4-.3.7-.3.7-.4 1-.6.5-.3 1-.3 1.2-.8.2-.6-.3-1.4-.3-1.7.4-2.8.9-4.1.2-5.8-.2-.3 1-4 1-3.4.2.9.3 3.8.3 5.9 0 .9.8 4.2 1 5.4.4 1.8-2.6 3.2-2 3.8l1 .3 1-.1c.3-.1.6-.5.7-.6.6-.8.4-.6 1-1.1.4-.5 0-1.4 0-1.8v-2.9c0-1.9 0-2.4.2-3.2l1-6.2s.2 0 .3-.2c.2-.2.1-.5.2-.7 0-.2 0 0 0 0l.2.6-.2.8c-.2.4.1.4.2.5 0 0 1-.7 1.2-1l.1-.8-.2-1.8c.2-.9 1-3.2.8-3.8Zm-2.2 3.6c-.2.5-.2 0-.6 1.7V31.1c0-2.1-1.5-6.5-.6-5.6 0 0 1.2.8 1.4 1.4.2.5.1 1.8-.2 3.4Z" data-v-d1ac2910=""></path>
            <path d="M20 35.3c-.3-.3-1.6-2-2.4-2.7l-1-1.5c-1.6-2-1.9-2.1-3-2.2-.1-.2 0-.3 0-.4 1.3-.5.9-1.6.8-1.7l.2-.6-.2-1-.1-.7c-.8-.9-2.6-.9-3.5-.3-.8.5-.3.9-.6 1.6-.1.5 0 .9.3 1.2 0 .3.8.7.7 1.2-.2.6-1 .7-1 .7l-.3.1c-1.4.8-2.7 2-2.9 3.2 0 1 1.7 2.5 2.7 3.3.1 1.7 1.2 4.4 1.3 4.5.1.3-.3-.3-1.4-.7-.6-.1-1.2-.9-1.4 0 0 .4 0 1-.3 1.8-.2.6 0 1.5.7 1.5.5 0 .8-1.4.8-1.4s2.3 1 3.3.7c.7-.2.8-2.7.2-5-.2-.8.4 1.3 2 3.1.2.4 0 .7.2 2.2v1c0 .7-.7 1.6.9 2 1.6.4 1.5.8 2.5.1.5-.3-1.4-.5-1.8-1.6v-.2c0-.9.3-2.9.2-3.6-.6-3.4-2-5-2.2-5.8 0-2.4-.2-3.2 0-2.6l1 1.5.3.3.4.4 2.8 2.1s.1 1 .6 1c.4 0 2.3-1 2.2-1.1 0-.3-2-.3-2-.4ZM9.7 34l-1-1.8c.4-.7 1-1 1-.9.2 0 0 2.7 0 2.7Z" data-v-d1ac2910=""></path>
            <path d="M51.8 54.8c-.1 0-1.2-.5-2.2-2-.5-.8-3.8-7-4.3-8.2-.9-2.4-8-6.7-8.2-7-.2-.1-.2-.5-.1-.8.5-2 .3-3.8.4-4.1.3-.7.8-1.7.4-2.3l-.7-1.2s3.7 1.4 4.6 1.2c.8-.2 4.5-2.8 4.7-2.9.5-.2 1.8.1 3-1.3.5-.5.2-.6 0-.6.1 0-.1-.2-.4 0-.2.1-1.3.4-1.1.2.8-.5.2-.7.2-.7-.3.3-1.8.7-2.1 1l-.8.5c-.9.5-2.4 1.3-4 1.7-.3 0-3.3-2-5-2.2-1-.1-1.3 0-1.8.1-.6.2-.6-.7 0-1 .2 0 .6-.1.7-.4.2-.4 0-.7 0-.8v-.2l-.1-.1v-.5c.3-.2.2-.4.1-.4-.1-.1-.6-.4-.7-.7 0-.6-.3-.9-.3-1.2 0 0-.1-.8-.4-1-.4-.4-1.2-.8-2.8-.3-.6.1-.5.4-1.8.6-1.6.3-2.4 2.7-2.3 3.7.1.8.8.4.8.7 0 .6-.1 1.5 1 1.8.3 0 1.8-.2 2.4 0 0 0 .5 0 .2.4-.1.2-1.6.9-2 1-1 .4-3.6 4.6-4.3 5-.7.3-1 .6-1.3.8-1 .6-2.1 1.4-2.7 1.4-.7 0-1.4.3-1.9 1.5-.1.3 0 .5.3.5 0 .1.2.3.4.3.2.2.5 0 .7 0l1-.7c.5-.2 4-1.5 4.7-2.1 1.2-.6 3.5-3 4-3s1.6 1 1.2 3c0 .7-1 2-1.4 2.6-1 1.5-.2 2.7-.4 3-1 1.6-2.2 3.7-4.2 5.5-.8.6-2.4.7-3.3 1-2 .5-4.2 2-5.7 2.8-.5.3-1-.3-1.6-.4-.7 0-.7 1-1 1.8l-.7 1.5c-.1.4-.7 1.7-.1 2.2.4.7 1.5-1.4 2.4-2.3.4-.5.7-.7 1.7-1 .6 0 5.6-1.4 8.8-2.2h.3c4.2-1.1 8.6-6.6 8.7-6.6l7.2 3.5c.7.4.7 1.8 1.5 3.4a22 22 0 0 0 4.4 4.9c.4.4 0 1 .2 1.6.2.6.6.6 1.1.6l1.6-.2c.7 0 .5 0 1.3.2.4 0 1.3.1 2-.5 1-1.2-1.2-.6-2.3-1.1Z" data-v-d1ac2910=""></path>
            <path d="M44 51.7a.4.4 0 0 0-.6-.2 22.2 22.2 0 0 1-21-.6.5.5 0 0 0-.6.2c-.1.2 0 .4.1.6a23 23 0 0 0 21.9.5c.2 0 .3-.3.2-.5Z" data-v-d1ac2910=""></path>
            <path d="M12.9 22.9h-.2c-.2-.2-.3-.4-.2-.6 2-4.4 5.2-8 9.3-10.4.2-.1.5 0 .6.2.2.2 0 .4-.1.6-4 2.3-7.1 5.7-9 10l-.4.2Z" data-v-d1ac2910=""></path>
            <path d="M50.3 17.3c-.1 0-.3 0-.3-.2A22 22 0 0 0 36.5 10c-.2 0-.4-.3-.3-.5 0-.3.2-.4.5-.4a23 23 0 0 1 14 7.5v.7h-.4Z" data-v-d1ac2910=""></path>
          </svg>
          <!--v-if-->
          <div data-v-a3ce97de="" if="serviceTitle">
            <div data-v-a3ce97de="" class="service-title"></div>
            <!--v-if-->
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</header>"
`;
